docs/environment-variables.md

```diff
@@ -55,6 +55,7 @@ This document outlines environment variables that can be used to customize behav
 | `WORKFLOW_GC_PERIOD`                     | `time.Duration`     | `5m`                                                                                        | The periodicity for GC of workflows.                                                                                                                                                                                                                                     |
 | `SEMAPHORE_NOTIFY_DELAY`                 | `time.Duration`     | `1s`                                                                                        | Tuning Delay when notifying semaphore waiters about availability in the semaphore                                                                                                                                                                                        |
 | `WATCH_CONTROLLER_SEMAPHORE_CONFIGMAPS` | `bool` | `true` | Whether to watch the Controller's ConfigMap and semaphore ConfigMaps for run-time changes. When disabled, the Controller will only read these ConfigMaps once and will have to be manually restarted to pick up new changes. |
+| `HOLDER_KEY_VERSION`                     | `string`            | `v1`                                                                                        | Whether to have the Holding field be populated using a reconstructed key (v1) or persistent key (v2)                       |
```

Comment:
These are in alphabetical order.
Also I don't think a user would have any idea how to interpret this 😕 "Holding field" is not really user-facing and "reconstructed" and "persistent" are not terms a user would understand either

Per above comment, I think you can just default to V2 always. The V1 detection is just for backward-compat with older workflows

Reply:
Yeah that is a valid point. Do you have any idea of how to phrase this?

I think we should always use V1, this changes how the workflow status is populated, it should be opt in at least for some time.